dbt_platform_helper/providers/copilot.py

```diff
@@ -0,0 +1,256 @@
+import json
+import random
+import string
+import time
+
+from botocore.exceptions import ClientError
+
+from dbt_platform_helper.providers.aws import AWSError
+from dbt_platform_helper.providers.aws import get_connection_secret_arn
+from dbt_platform_helper.providers.aws import (
+    get_postgres_connection_data_updated_with_master_secret,
+)
+from dbt_platform_helper.utils.application import Application
+from dbt_platform_helper.utils.messages import abort_with_error
+
+# TODO move to constants
+CONDUIT_DOCKER_IMAGE_LOCATION = "public.ecr.aws/uktrade/tunnel"
+CONDUIT_ADDON_TYPES = [
+    "opensearch",
+    "postgres",
+    "redis",
+]
+
+
+# TODO exceptions?
+class NoClusterError(AWSError):
+    pass
+
+
+class CreateTaskTimeoutError(AWSError):
+    pass
+
+
+class ParameterNotFoundError(AWSError):
+    pass
+
+
+class AddonNotFoundError(AWSError):
+    pass
+
+
+class InvalidAddonTypeError(AWSError):
+    def __init__(self, addon_type):
+        self.addon_type = addon_type
+
+
+def get_addon_type(ssm_client, application_name: str, env: str, addon_name: str) -> str:
+    addon_type = None
+    try:
+        addon_config = json.loads(
+            ssm_client.get_parameter(
+                Name=f"/copilot/applications/{application_name}/environments/{env}/addons"
+            )["Parameter"]["Value"]
+        )
+    except ssm_client.exceptions.ParameterNotFound:
+        raise ParameterNotFoundError
+
+    if addon_name not in addon_config.keys():
+        raise AddonNotFoundError
+
+    for name, config in addon_config.items():
+        if name == addon_name:
+            addon_type = config["type"]
+
+    if not addon_type or addon_type not in CONDUIT_ADDON_TYPES:
+        raise InvalidAddonTypeError(addon_type)
+
+    if "postgres" in addon_type:
+        addon_type = "postgres"
+
+    return addon_type
+
+
+# TODO ECS method
+def get_cluster_arn(ecs_client, application_name: str, env: str) -> str:
+
+    # TODO refactor
```

Comment:
I'll update line 77 todo as it has to do with the functionality, otherwise the line 74 TODO is addressed in https://github.com/uktrade/platform-tools/tree/DBTP-1520-provider-split will leave as is for now.

Reply:
updated